### PR TITLE
Clarify 'non-exhaustive list of the values defined by this specification'

### LINKED ISSUE #138: "A non-exhaustive list of valid values defined by this specification are" is weird language
The current draft contains phrases about "non-exhaustive list of valid values defined by this specification" in a few places:

>  non-exhaustive list of valid values defined by this specification are string, number, and image media types such as image/jpeg as defined in IANA media type registry for images 

(e.g. https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#appendix-E.2.2 )

> To express the specifics about the claim, the most deeply nested value MAY be an object that includes a following non-exhaustive list of parameters defined by this specification:

(e.g.  https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#appendix-E.1.1.2 )

In both cases I've not been able to find an exhaustive list of the values that the specification does actually define. I'm not sure exactly what is meant, in other parts of the specification language like "Below is a non-exhaustive list of valid parameters that MAY be included" is used which may be the intention, or perhaps it is more like "This specification defines the following parameters, other parameters may be used and implementations MUST ignore unrecognised parameters"?

### PR DESCRIPTION
**Pull request by jogu**
closes #138

### CODE REVIEW

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 1]
```diff
@@ -1874,9 +1874,9 @@ The following additional Credential Issuer metadata parameters are defined for t
 
 * `credential_definition`: REQUIRED. Object containing the detailed description of the Credential type. It consists of at least the following two parameters:
   * `type`: REQUIRED. Array designating the types a certain Credential type supports, according to [@VC_DATA], Section 4.3.
-  * `credentialSubject`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes a following non-exhaustive list of parameters defined by this specification:
+  * `credentialSubject`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes the following parameters (other parameters MAY also be used):
```
Comment (selfissued) [inline openid-4-verifiable-credential-issuance-1_0.md:1877]:
```suggestion
  * `credentialSubject`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes the following parameters. Other parameters MAY also be used.
```

Comment (Sakurann) [reply openid-4-verifiable-credential-issuance-1_0.md:1877]: please add `the following parameters defined by this specification. Other...`

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 2]
```diff
@@ -1874,9 +1874,9 @@ The following additional Credential Issuer metadata parameters are defined for t
...
       * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim, and/or if the Credential Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default value is `false`.
-      * `value_type`: OPTIONAL. String value determining the type of value of the claim. A non-exhaustive list of valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+      * `value_type`: OPTIONAL. String value determining the type of value of the claim. Values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```
Comment (Sakurann) [inline openid-4-verifiable-credential-issuance-1_0.md:1879]:
```suggestion
      * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 3]
```diff
@@ -1937,9 +1937,9 @@ The following additional Credential Issuer metadata parameters are defined for t
 * `credential_definition`: REQUIRED. Object containing the detailed description of the Credential type. It consists of at least the following three parameters:
   * `@context`: REQUIRED. Array as defined in [@VC_DATA], Section 4.1.
   * `type`: REQUIRED. Array designating the types a certain credential type supports, according to [@VC_DATA], Section 4.3.
-  * `credentialSubject`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes a following non-exhaustive list of parameters defined by this specification:
+  * `credentialSubject`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes the parameters (other parameters MAY also be included):
```
[the remark this answers is not in the document]
Comment (Sakurann) [reply openid-4-verifiable-credential-issuance-1_0.md:1940]: please add `the following parameters defined by this specification. Other...`

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 4]
```diff
@@ -1937,9 +1937,9 @@ The following additional Credential Issuer metadata parameters are defined for t
...
       * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim, and/or if the Credential Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default value is `false`.
-      * `value_type`: OPTIONAL. String value determining the type of value of the claim. A non-exhaustive list of valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+      * `value_type`: OPTIONAL. String value determining the type of value of the claim. Values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```
Comment (Sakurann) [inline openid-4-verifiable-credential-issuance-1_0.md:1942]:
```suggestion
      * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 5]
```diff
@@ -2030,7 +2030,7 @@ The following additional Credential Issuer metadata parameters are defined for t
...
     * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim, and/or if the Credential Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default value is `false`.
-    * `value_type`: OPTIONAL. String value determining the type of value of the claim. A non-exhaustive list of valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+    * `value_type`: OPTIONAL. String value determining the type of value of the claim. Values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```
Comment (Sakurann) [inline openid-4-verifiable-credential-issuance-1_0.md:2033]:
```suggestion
    * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 6]
```diff
@@ -2080,9 +2080,9 @@ The following additional Credential Issuer metadata parameters are defined for t
 
 
 * `vct`: REQUIRED. String designating the type of a Credential, as defined in [@!I-D.ietf-oauth-sd-jwt-vc].
-* `claims`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim about the subject offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes a following non-exhaustive list of parameters defined by this specification:
+* `claims`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim about the subject offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes the following parameters:
```
Comment (Sakurann) [inline openid-4-verifiable-credential-issuance-1_0.md:2083]:
```suggestion
* `claims`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim about the subject offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes the following parameters defined by this specification:
```

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 7]
```diff
@@ -2080,9 +2080,9 @@ The following additional Credential Issuer metadata parameters are defined for t
...
     * `mandatory`: OPTIONAL. Boolean which when set to `true` indicates the claim MUST be present in the issued Credential. If the `mandatory` property is omitted its default should be assumed to be `false`.
-    * `value_type`: OPTIONAL. String value determining the type of value of the claim. A non-exhaustive list of valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+    * `value_type`: OPTIONAL. String value determining the type of value of the claim. Values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```
Comment (Sakurann) [inline openid-4-verifiable-credential-issuance-1_0.md:2085]:
```suggestion
    * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```

**Review by Sakurann — CHANGES_REQUESTED** [review]
the current language conveys two messages:
1. this spec defines valid values
2. other values can be used.
If the language is being changed, please include point 1 - I made suggestions.

File: openid-4-verifiable-credential-issuance-1_0.md (hunk 1 above)
Comment (jogu) [reply openid-4-verifiable-credential-issuance-1_0.md:1877]: @selfissued @Sakurann  Are we definitely okay removing the ':'? This is the start of a list and everywhere else in the spec we use a colon in this way.

Comment (Sakurann) [reply openid-4-verifiable-credential-issuance-1_0.md:1877]: oh, umm let's keep `:` probably.

File: openid-4-verifiable-credential-issuance-1_0.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jogu) [reply openid-4-verifiable-credential-issuance-1_0.md:1940]:
```suggestion
  * `credentialSubject`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes the following parameters defined by this specification (other parameters MAY also be included):
```

File: openid-4-verifiable-credential-issuance-1_0.md (hunk 1 above)
Comment (jogu) [reply openid-4-verifiable-credential-issuance-1_0.md:1877]:
```suggestion
  * `credentialSubject`: OPTIONAL. Object containing a list of name/value pairs, where each name identifies a claim offered in the Credential. The value can be another such object (nested data structures), or an array of such objects. To express the specifics about the claim, the most deeply nested value MAY be an object that includes the following parameters defined by this specification (other parameters MAY also be used):
```

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 8]
```diff
@@ -1874,9 +1874,9 @@ The following additional Credential Issuer metadata parameters are defined for t
...
       * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim, and/or if the Credential Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default value is `false`.
-      * `value_type`: OPTIONAL. String value determining the type of value of the claim. A non-exhaustive list of valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+      * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```
Comment (jogu) [inline openid-4-verifiable-credential-issuance-1_0.md:1879]:
```suggestion
      * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 9]
```diff
@@ -1937,9 +1937,9 @@ The following additional Credential Issuer metadata parameters are defined for t
...
```
Comment (jogu) [inline openid-4-verifiable-credential-issuance-1_0.md:1942]:
```suggestion
      * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 10]
```diff
@@ -2030,7 +2030,7 @@ The following additional Credential Issuer metadata parameters are defined for t
...
     * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the Credential Issuer will always include this claim in the issued Credential. If set to `false`, the claim is not included in the issued Credential if the wallet did not request the inclusion of the claim, and/or if the Credential Issuer chose to not include the claim. If the `mandatory` parameter is omitted, the default value is `false`.
-    * `value_type`: OPTIONAL. String value determining the type of value of the claim. A non-exhaustive list of valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+    * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```
Comment (jogu) [inline openid-4-verifiable-credential-issuance-1_0.md:2033]:
```suggestion
    * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```

File: openid-4-verifiable-credential-issuance-1_0.md [hunk 11]
```diff
@@ -2080,9 +2080,9 @@ The following additional Credential Issuer metadata parameters are defined for t
...
     * `mandatory`: OPTIONAL. Boolean which when set to `true` indicates the claim MUST be present in the issued Credential. If the `mandatory` property is omitted its default should be assumed to be `false`.
-    * `value_type`: OPTIONAL. String value determining the type of value of the claim. A non-exhaustive list of valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+    * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```
Comment (jogu) [inline openid-4-verifiable-credential-issuance-1_0.md:2085]:
```suggestion
    * `value_type`: OPTIONAL. String value determining the type of value of the claim. Valid values defined by this specification are `string`, `number`, and image media types such as `image/jpeg` as defined in IANA media type registry for images (https://www.iana.org/assignments/media-types/media-types.xhtml#image). Other values MAY also be used.
```

**Comment by jogu** [comment]
I've put back all the 'defined by this specification'. As per discussion I've not made the changes that required removed the ':'s. This should be good to go now.

**Review by Sakurann — APPROVED** [review]

**Comment by Sakurann** [comment]
lgtm. will merge once 1-2 more approvals kick in.

**Review by bc-pi — APPROVED** [review]

**Pull request merged**